gov2/iam/actions/users.go

```diff
@@ -0,0 +1,227 @@
+	}
+	return user, err
+}
+
+// snippet-end:[gov2.iam.CreateUser]
+
+// snippet-start:[gov2.iam.PutUserPolicy]
+
+// CreateUserPolicy adds an inline policy to a user. This example creates a policy that
+// grants a list of actions on a specified role.
+// PolicyDocument shows how to work with a policy document as a data structure and
+// serialize it to JSON by using Go's JSON marshaler.
+func (wrapper UserWrapper) CreateUserPolicy(userName string, policyName string, actions []string,
+		roleArn string) error {
+	policyDoc := PolicyDocument{
+		Version:   "2012-10-17",
+		Statement: []PolicyStatement{{
+			Effect: "Allow",
+			Action: actions,
+			Resource: aws.String(roleArn),
+		}},
+	}
+	policyBytes, err := json.Marshal(policyDoc)
+	if err != nil {
+		log.Printf("Couldn't create policy document for %v. Here's why: %v\n", roleArn, err)
+		return err
+	}
+	_, err = wrapper.IamClient.PutUserPolicy(context.TODO(), &iam.PutUserPolicyInput{
+		PolicyDocument: aws.String(string(policyBytes)),
+		PolicyName:     aws.String(policyName),
+		UserName:       aws.String(userName),
+	})
+	if err != nil {
+		log.Printf("Couldn't create policy for user %v. Here's why: %v\n", userName, err)
+	}
+	return err
+}
+
+// snippet-end:[gov2.iam.PutUserPolicy]
+
+// snippet-start:[gov2.iam.ListUserPolicies]
+
+// ListUserPolicies lists the inline policies for the specified user.
+func (wrapper UserWrapper) ListUserPolicies(userName string) ([]string, error) {
+	var policies []string
+	result, err := wrapper.IamClient.ListUserPolicies(context.TODO(), &iam.ListUserPoliciesInput{
+		UserName: aws.String(userName),
+	})
+	if err != nil {
+		log.Printf("Couldn't list policies for user %v. Here's why: %v\n", userName, err)
+	} else {
+		policies = result.PolicyNames
+	}
+	return policies, err
+}
+
+// snippet-end:[gov2.iam.ListUserPolicies]
+
+// snippet-start:[gov2.iam.DeleteUserPolicy]
+
+// DeleteUserPolicy deletes an inline policy from a user.
+func (wrapper UserWrapper) DeleteUserPolicy(userName string, policyName string) error {
+	_, err := wrapper.IamClient.DeleteUserPolicy(context.TODO(), &iam.DeleteUserPolicyInput{
+		PolicyName: aws.String(policyName),
+		UserName:   aws.String(userName),
+	})
+	if err != nil {
+		log.Printf("Couldn't delete policy from user %v. Here's why: %v\n", userName, err)
+	}
+	return err
+}
+
+// snippet-end:[gov2.iam.DeleteUserPolicy]
+
+// snippet-start:[gov2.iam.DeleteUser]
+
+// DeleteUser deletes a user.
+func (wrapper UserWrapper) DeleteUser(userName string) error {
+	_, err := wrapper.IamClient.DeleteUser(context.TODO(), &iam.DeleteUserInput{
+		UserName: aws.String(userName),
+	})
+	if err != nil {
+		log.Printf("Couldn't delete user %v. Here's why: %v\n", userName, err)
+	}
+	return err
+}
+
+// snippet-end:[gov2.iam.DeleteUser]
+
+// snippet-start:[gov2.iam.CreateAccessKey]
+
+// CreateAccessKeyPair creates an access key for a user. The returned access key contains
+// the pair of ID and secret credentials needed to use the key.
```

Comment:
pair of ID and secret credentials needed --> ID and secret credentials that are needed

Reply:
fixed